Staffing level endpoint

johnschoeman opened the pull request:
This creates an endpoint that responds with the current Event staffing level that will be used for the alert indicator device.

It is scoped under an api namespace to keep it seperate from the main application.

making a get request to ```/api/staffinglevel``` should return some json of the form: 
```
{staffing_level_number: 3, staffing_level_name: "Satisfied", staffing_level_percentage: 80}
```

jbuilder was added to the gemfile to help with the structuring of the json data.

Testing for the Event model was added in ```spec/features/alert_spec.rb```

Note that there is currently no authentication happening at the controller level.  I think this is okay for now since the endpoint doesn't allow for any mutations.  Authentication can be added later after the IoT end of this is more fully worked out.

kgf commented:
@johnschoeman We have merged the Rails 5 code into development. Could you please re-point this PR towards development ?
Thanks for your help.
Kevin

johnschoeman committed:
[staffing-level-endpoint] Create staffing level api endpoint

Why:
We would like to make the current staffing level information available
in an api endpoint for consumption by alert indicators.

This PR:
Introduces and endpoint which will return some JSON indicting the current
staffing level. A new namespace, api, was added to the routes to
separate this concern from the rest of the app. jbuilder was added as a
gem to use as the json serializer/templater.

johnschoeman commented:
@kgf rebased onto development.  It would probably make sense to revisit the api and implementation as well.

johnschoeman commented:
Especially since it's not passing any ci checks

kgf commented:
So, should we close this PR ? Will you (hopefully) open a new one ?

kgf commented:
This PR is of more interest. Will you be able to continue with it ?

johnschoeman committed:
[staffing-level-endpoint] Continue with refactor

johnschoeman committed:
[staffing-level-endpoint] fix tests

johnschoeman committed:
[staffing-level-endpoint] fix request spec

johnschoeman commented:
@kgf , I went over the code again here and cleaned up some stuff, added a few tests.

I think it is good to rebase and merge. We may want to add some better factories and add an ssh token or something to handle basic auth directly after merging this.  

Sorry it took me a while to get back around to this by the way.  I should have a bit more availability moving forward.

kgf committed:
Merge branch 'development' into staffing-level-endpoint

kgf committed:
Merge branch 'development' into staffing-level-endpoint

gasparyn commented on app/models/event.rb:
```diff
@@ -163,11 +171,31 @@ def previous_event
+
   def calc_duration #This is also used in timecards; it should be extracted out
-     if !(start_time.blank?) and !(end_time.blank?)
+    if !(start_time.blank?) and !(end_time.blank?)
```
The `and` here might give us unintended consequences because precedence. 
Additionally, it's better to use `if start_time.present? && end_time.present?`

gasparyn commented on app/models/staffing_level.rb:
```diff
@@ -0,0 +1,48 @@
+  end
+
+  def current_or_next_events
+    if !current_events.empty?
```
I'd do something similar here:
`if current_events.present?`

gasparyn commented on app/models/staffing_level.rb:
```diff
@@ -0,0 +1,48 @@
+  def current_or_next_events
+    if !current_events.empty?
+      current_events
+    elsif !next_event.empty?
```
similarly: `elsif next_event.present?`
